Reqnroll.VisualStudio/Editor/Services/UnusedStepDefinitionsFinder.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Reqnroll.VisualStudio.ProjectSystem;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+#nullable disable
+namespace Reqnroll.VisualStudio.Editor.Services;
+
+public class UnusedStepDefinitionsFinder
+{
+    private readonly IIdeScope _ideScope;
+
+    public UnusedStepDefinitionsFinder(IIdeScope ideScope)
+    {
+        _ideScope = ideScope;
+    }
+
+    public IEnumerable<ProjectStepDefinitionBinding> FindUnused(ProjectStepDefinitionBinding[] stepDefinitions,
+                                                        string[] featureFiles, DeveroomConfiguration configuration)
+    {
+        return featureFiles.SelectMany(ff => FindUnused(stepDefinitions, ff, configuration));
+    }
+
+    public IEnumerable<ProjectStepDefinitionBinding> FindUnused(ProjectStepDefinitionBinding[] stepDefinitions, 
+        string featureFilePath, DeveroomConfiguration configuration) => 
+        LoadContent(featureFilePath, out string featureFileContent)
+        ? FindUsagesFromContent(stepDefinitions, featureFileContent, featureFilePath, configuration)
+        : Enumerable.Empty<ProjectStepDefinitionBinding>();
+
+    private IEnumerable<ProjectStepDefinitionBinding> FindUsagesFromContent(ProjectStepDefinitionBinding[] stepDefinitions, string featureFileContent, string featureFilePath, DeveroomConfiguration configuration)
+    {
+        var dialectProvider = ReqnrollGherkinDialectProvider.Get(configuration.DefaultFeatureLanguage);
+        var parser = new DeveroomGherkinParser(dialectProvider, _ideScope.MonitoringService);
+        parser.ParseAndCollectErrors(featureFileContent, _ideScope.Logger,
+            out var gherkinDocument, out _);
+
+        var featureNode = gherkinDocument?.Feature;
+        if (featureNode == null)
+            return Enumerable.Empty<ProjectStepDefinitionBinding>();
+
+        var dummyRegistry = ProjectBindingRegistry.FromBindings(stepDefinitions);
+
+        //this set keeps track of which step definitions have not been used. We start with ALL step definitions, and will subtract those that match steps in the feature file
+        HashSet<ProjectStepDefinitionBinding> stepDefinitionsSet = new(stepDefinitions);
```

Comment:
Following the comment above, here you can use the `bindingRegistry.StepDefinitions` instead of the `stepDefinitions`